src/map/mob.c

```diff
@@ -521,76 +551,100 @@ static struct mob_data *mob_once_spawn_sub(struct block_list *bl, int16 m, int16
 	else
 		strcpy(data.name, DEFAULT_MOB_JNAME);
 
-	if (event)
+	if (event != NULL)
 		safestrncpy(data.eventname, event, sizeof(data.eventname));
 
-	// Locate spot next to player.
-	if (bl && (x < 0 || y < 0))
+	/** Locate spot next to player. **/
+	if (bl != NULL && (x < 0 || y < 0))
 		map->search_freecell(bl, m, &x, &y, 1, 1, 0);
 
-	// if none found, pick random position on map
+	/** If none found, pick random position on map. **/
 	if (x <= 0 || x >= map->list[m].xs || y <= 0 || y >= map->list[m].ys)
 		map->search_freecell(NULL, m, &x, &y, -1, -1, 1);
 
 	data.x = x;
 	data.y = y;
 
-	if (!mob->parse_dataset(&data))
+	if (mob->parse_dataset(&data) == 0)
 		return NULL;
 
-	return mob->spawn_dataset(&data);
+	return mob->spawn_dataset(&data, npc_id);
 }
 
-/*==========================================
- * Spawn a single mob on the specified coordinates.
- *------------------------------------------*/
-static int mob_once_spawn(struct map_session_data *sd, int16 m, int16 x, int16 y, const char *mobname, int class_, int amount, const char *event, unsigned int size, unsigned int ai)
+/**
+ * Spawns a given amount of mobs.
+ *
+ * @param sd The invoking character.
+ * @param m The ID of the map where the mob should be spawned.
+ * @param x The x coordinate where the mob should be spawned.
+ * @param y The y coordinate where the mob should be spawned.
+ * @param mobname The mob's display name.
+ * @param class_ The mob's ID in database.
+ * @param amount The amount of mobs to spawn.
+ * @param event The name of the event which should be executed when the mob is killed.
+ * @param size The mob's size.
+ * @param ai The mob's AI.
+ * @return The last spawned mob's GID, or 0 if spawning failed.
+ *
+ **/
+static int mob_once_spawn(struct map_session_data *sd, int16 m, int16 x, int16 y, const char *mobname, int class_,
+			  int amount, const char *event, unsigned int size, unsigned int ai)
 {
-	struct mob_data* md = NULL;
-	int count, lv;
 	bool no_guardian_data = false;
 
-	if( ai && ai&0x200 ) {
+	if (ai > 0 && (ai & 0x200) == 0x200) {
 		no_guardian_data = true;
-		ai &=~ 0x200;
+		ai &= ~0x200;
 	}
 
 	if (m < 0 || amount <= 0)
-		return 0; // invalid input
+		return 0;
 
-	lv = (sd) ? sd->status.base_level : 255;
+	struct mob_data *md = NULL;
+
+	for (int i = 0; i < amount; i++) {
+		int mob_id = class_;
+
+		if (mob_id < 0) {
+			mob_id = mob->get_random_id(-class_ - 1, (battle_config.random_monster_checklv == 1) ? 3 : 1,
```

Comment:
weird, some `mob->get_random_id(...)` are using max lv 99 like 
```
class_ = mob->get_random_id(-class_-1, 1, 99);
```
but some are using max lv 255 ... perhaps mistaken from ancient code changes?

Reply:
Well, I can't tell why or when this happened, but it looks weird. I'll have a look at it.
Since this isn't related to the PR's subject, I'd ask you to create an issue or a topic at our board.